### ISSUE TITLE
Fragile Invoice matching

### ISSUE THREAD

**Issue by artur-beljajev** [internetee/registry#1421]
Currently we expect only one number to be present in the payment description. I find this assumption wrong.

```
irb(main):001:0> bank_transaction = BankTransaction.first
  BankTransaction Load (1.2ms)  SELECT  "bank_transactions".* FROM "bank_transactions"  ORDER BY "bank_transactions"."id" ASC LIMIT 1
=> #<BankTransaction id: 980190962, bank_statement_id: nil, bank_reference: nil, iban: "US75512108001245126199", currency: "EUR", buyer_bank_code: nil, buyer_iban: nil, buyer_name: nil, document_no: nil, description: "RKID:20191114000197283537 RIIGI INFOKOMMUNIKATSIOO...", sum: 0.1e1, reference_no: nil, paid_at: nil, created_at: "2019-10-11 07:12:11", updated_at: "2019-10-11 07:12:11", creator_str: nil, updator_str: nil>
irb(main):002:0> bank_transaction.description
=> "RKID:3456345634563456 FOOBAR INC SA Arve: 134408"
irb(main):003:0> bank_transaction.invoice_num
=> 3456345634563456
```

**Comment by artur-beljajev** [comment]
Related: https://github.com/internetee/registry/issues/1224

**Comment by vohmar** [comment]
to be exact we expect invoice nr being the first numeric value in description field and this is communicated as requirement to registrars as well. Unfortunately people forget. There are no other limitations on the description content nor number of numerical values in it. But as you already referenced (#1224 ) we have planned to change the current invoice matching logic so that this specific problem here is resolved as well.

**Comment by artur-beljajev** [comment]
> Unfortunately people forget

Exactly. And I think this is just a fact that should be taken into account. I believe it's naive to keep fingers crossed that everybody will follow this pattern.

#1224 doesn't seem to above mentioned problem. We need to rethink description parsing algorithm anyway.

**Comment by vohmar** [comment]
#1224 removes the invoice number from the equation all together we sill only search for reference nr on refnr field or if this is missing from description field and then use transferred amount to find match among open invoices for that registrar. Yes this requires changing parsing algorithm.

**Comment by artur-beljajev** [comment]
#1224 states:

>  where only **one** of the above mentioned properties is provided.

It says nothing that _only_ reference number will be matched. The logic you described in your previous comment differs from that is described in #1224.

**Comment by vohmar** [comment]
now I understand what you meant. I actually has this ticket in mind: https://github.com/internetee/registry/issues/1415
